core/input/input.cpp

```diff
@@ -77,6 +77,8 @@ Input *Input::singleton = nullptr;
 
 void (*Input::set_mouse_mode_func)(Input::MouseMode) = nullptr;
 Input::MouseMode (*Input::get_mouse_mode_func)() = nullptr;
+void (*Input::set_mouse_mode_override_func)(Input::MouseMode) = nullptr;
+void (*Input::set_mouse_mode_override_enabled_func)(bool) = nullptr;
 void (*Input::warp_mouse_func)(const Vector2 &p_position) = nullptr;
 Input::CursorShape (*Input::get_current_cursor_shape_func)() = nullptr;
 void (*Input::set_custom_mouse_cursor_func)(const Ref<Resource> &, Input::CursorShape, const Vector2 &) = nullptr;
```

Comment:
Is there any reason why these are using separate function pointers and wrappers in the `DisplayServer`, instead of calling `DisplayServer::get_singleton()->mouse_???_mode`?

Reply:
That's a great question.

Reply:
I guess this might be to avoid including headers from the `servers` in the `core`.